Rework Riemann event construction

jarpy opened the pull request:
The previous implementation had problems when the "map_fields" and
"riemann_event" configuration options compete to set the value of
a Riemann event field. "riemann_event" should win, but was not winning.

Additionally, casting operations during event contruction could result
in strings being sent to Riemann where numbers are required. The "ttl"
field, in particular, had this problem.

Fixes: #22
Fixes: #23

Commit:
Rework Riemann event construction

The previous implementation had problems when the "map_fields" and
"riemann_event" configuration options competed to set the value of
a Riemann event field. "riemann_event" should win, but was not winning.

Additionally, casting operations during event contruction could result
in strings being sent to Riemann where numbers are required. The "ttl"
field, in particular, had this problem.

Fixes: #22
Fixes: #23

jsvd reviewed (approved):
LGTM, thanks for this change Toby!
can you also bump the patch version in the gemspec and add a changelog entry noting the fixing of these two issues so I can publish the plugin asap?

Commit:
Fix riemann_event not overriding and ensure correct types

Fix values from "riemann_event" not overwriting those from "map_fields".
Closes: #22

Fix ttl, metric sometimes being sent as string, not float.
Closes: #23

ghost commented:
With pleasure!

Thanks so much for the quick review, João.

jsvd commented:
Version 3.0.6 has been published! https://rubygems.org/gems/logstash-output-riemann/versions/3.0.6
